### ISSUE TITLE
[LOVE] Test out the etcd fs implementation

### ISSUE THREAD

**Issue by purpleidea** [purpleidea/mgmt#309]
#### Welcome:

This is a intermediate level patch for someone who is learning or somewhat knowledgeable about golang, and is new to mgmt. It is meant to encourage new programmers to get involved with mgmt. It will probably take you a few hours to a day or so to complete.

#### Intro:

Mgmt has runs an embedded etcd server as part of it's clustering operations. To deploy code into etcd, the user uses an mgmt "deploy" utility to bundle a source directory of code and files which are then stored into etcd.
In order to store a filesystem in etcd, there is an `etcdfs` layer which was written. It needs proper testing.
To implement this, you'll need to either recommend an alternate design, or use the one provided below.

#### Design:

The etcdfs layer lives in the `etcd/fs/` directory. Currently there is a stub of a test in `fs_test.go`.
It currently does _not_ spin up an etcd server to test, but it should. You should first write a test helper method to spin up an mgmt binary.
You should then patch `fs_test.go` so that it uses this helper and appropriately errors on failure instead of just logging.
Lastly you should write some number of filesystem tests. Tests can also go in `file_test.go` or in a separate file entirely if you'd like to write some more complex tests which belong elsewhere.

Read the contributing tips:
https://github.com/purpleidea/mgmt/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Happy hacking, and ping me for questions!

#### Bonus/Future work:

* The test helper could support spinning up a cluster of `N` mgmt binaries.
* Write a bunch of filesystem tests.
* Write some tests that run in parallel to test for races in the filesystem implementation. (Probably not necessary for our project, but always nice to have.)

**Comment by hexfusion** [comment]
Hi @purpleidea, I would be interested in picking this up. Thank you for the nice outline of the tasks.

**Comment by purpleidea** [comment]
@hexfusion Fantastic! Assuming you're going to start with some basic tests, great and let me know what you find. I hope their aren't any bugs, but no promises.

If you end up in "clustered tests" scenarios, please ping before writing a whole integration test helper. There's someone working on such a thing, which combining forces on would be preferable :)

Thanks!

**Comment by sushsada** [comment]
Is this task done?
"Design" says 

> It currently does not spin up an etcd server to test, but it should. You should first write a test helper method to spin up an mgmt binary.

But current file looks like it spins up an etcd server at:
https://github.com/purpleidea/mgmt/blob/443f489152ac7b9808bbf19f5004f63606286a0a/etcd/fs/fs_test.go#L75
